File: src/CorporateMessengerWithAgile/Application/Dto/Summary/CompanySummaryDto.cs
namespace Application.Dto.Summary
{
    public class CompanySummaryDto : BaseDto
    {
        public string Title { get; set; } = null!;
        public IReadOnlyList<Guid> EmployeeIds { get; set; } = null!;
        public IReadOnlyList<Guid> PositionIds { get; set; } = null!;
        public IReadOnlyList<Guid> ProjectIds { get; set; } = null!;
    }
}

File: src/CorporateMessengerWithAgile/ReactApp/reactapp.client/src/models/entity/CompanySummaryDto.ts
// src/models/entity/CompanySummaryDto.ts
import { z } from 'zod';
import { BaseDtoSchema } from './BaseDto';
import { GuidSchema } from '../Guid';

export const CompanySummaryDtoSchema = BaseDtoSchema.extend({
    title: z.string().min(1),
    employeeIds: z.array(GuidSchema).optional(),
    positionIds: z.array(GuidSchema).optional(),
    projectIds: z.array(GuidSchema).optional(),
});

export type CompanySummaryDto = z.infer<typeof CompanySummaryDtoSchema>;
